loopy/target/c/__init__.py

```diff
@@ -825,16 +828,23 @@ def get_function_definition(
                         tv.initializer is not None):
                     assert tv.read_only
 
-                    decl: Generable = self.wrap_global_constant(
+                    decl = self.wrap_global_constant(
                             self.get_temporary_var_declarator(codegen_state, tv))
 
                     if tv.initializer is not None:
-                        decl = Initializer(decl, generate_array_literal(
+                        init_decl = Initializer(decl, generate_array_literal(
```

Comment:
This is necessary because `Initializer` takes a `Declarator`, not a `Generable` (like the forced annotation above).